iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,320 @@
+// limitations under the License.
+#include "iceoryx_posh/internal/roudi_environment/roudi_environment.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::runtime;
+using iox::roudi::RouDiEnvironment;
+class PoshRuntime_test : public Test
+{
+  public:
+    PoshRuntime_test()
+    {
+    }
+
+    virtual ~PoshRuntime_test()
+    {
+    }
+
+    virtual void SetUp(){};
+    virtual void TearDown(){};
+
+    void InterOpWait()
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(200));
+    }
+
+    iox::cxx::GenericRAII m_errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [](const iox::Error, const std::function<void()>, const iox::ErrorLevel) { m_errorHandlerCalled = true; });
+
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    PoshRuntime* m_receiverRuntime{&iox::runtime::PoshRuntime::getInstance("/receiver")};
+    PoshRuntime* m_senderRuntime{&iox::runtime::PoshRuntime::getInstance("/sender")};
+    MqMessage m_sendBuffer;
+    MqMessage m_receiveBuffer;
+    const iox::cxx::CString100 m_runnableName = iox::cxx::CString100("testRunnable");
+    const iox::cxx::CString100 m_invalidRunnableName = iox::cxx::CString100("invalidRunnable,");
+    static bool m_errorHandlerCalled;
+};
+
+bool PoshRuntime_test::m_errorHandlerCalled{false};
+
+
+TEST_F(PoshRuntime_test, AppnameLength_TooLong)
+{
+    const std::string string104chars =
+        "/MXIYXHyPF9KjXAPv9ev9jxofYDArZzTvf8FF5uaWWC4dwabcjW75DurqeN645IabAsXVfngor7784446vb4vhArwBxLZlN1k1Qmrtz";
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(string104chars); },
+                 "Application name has more than 100 characters, including null termination!");
+}
+
+
+TEST_F(PoshRuntime_test, AppnameLength_ok)
+{
+    const std::string string100chars =
+        "/MXIYXHyPF9KjXAPv9ev9jxofYDArZzTvf8FF5uaWWC4dwabcjW75DurqeN645IabAsXVfngor7784446vb4vhArwBxLZlN1k1";
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntime::getInstance(string100chars); });
+}
+
+
+TEST_F(PoshRuntime_test, NoAppname)
+{
+    const std::string wrong("");
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(wrong); },
+                 "Cannot initialize runtime. Application name must not be empty!");
+}
+
+
+TEST_F(PoshRuntime_test, NoLeadingSlash_Appname)
+{
+    const std::string wrong = "wrongname";
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(wrong); },
+                 "Cannot initialize runtime. Application name wrongname does not have the required leading slash '/'");
+}
+
+
+// test class creates instance of Poshruntime, so when getInstance() is called without name it reuturns extisting
+// instance
+TEST_F(PoshRuntime_test, DISABLED_AppnameEmpty)
+{
+    EXPECT_DEATH({ iox::runtime::PoshRuntime::getInstance(); },
+                 "Cannot initialize runtime. Application name has not been specified!");
+}
+
+
+TEST_F(PoshRuntime_test, GetInstanceName)
```

Comment:
What are you testing here ? The test name should explain the test well enough so if it is seen in an error report we know immediately what is failing.